src/main/recorder.ts

```diff
@@ -280,6 +289,52 @@ type RecorderOptionsType = {
         obsRecorder.reconfigure(this._options);
         if (mainWindow) mainWindow.webContents.send('refreshState');
     }
+
+    /**
+     * Generate a filename for the final video of a recording according
+     * to its category, if possible.
+     *
+     * The final filename should contain adequate information to quickly be able
+     * to identify the video files on the filesystem and be able to tell what it
+     * was about and the result of it.
+     */
+    getFinalVideoFilename (metadata: Metadata): string | undefined {
+        let outputFilename = '';
+        let zoneName: string;
+        const resultText = getVideoResultText(metadata.category, metadata.result);
+
+        switch (metadata.category) {
+            case VideoCategory.TwoVTwo:
+            case VideoCategory.ThreeVThree:
+            case VideoCategory.Skirmish:
+            case VideoCategory.SoloShuffle:
+                zoneName = getInstanceNameByZoneId(metadata.zoneID);
+                outputFilename = `${zoneName} (${resultText})`;
+            break;
+
+            case VideoCategory.Raids:
+                const encounterName = getEncounterNameById(metadata.encounterID);
+                zoneName = getRaidNameByEncounterId(metadata.encounterID);
+                outputFilename = `${zoneName}, ${encounterName} (${resultText})`;
+            break;
+
+            case VideoCategory.MythicPlus:
+                if (metadata.challengeMode) {
+                    const cm = metadata.challengeMode;
+                    const keystoneUpgradeLevel = ChallengeModeDungeon.calculateKeystoneUpgradeLevel(cm.allottedTime, cm.duration);;
+                    const resultText = cm?.timed ? '+' + keystoneUpgradeLevel : 'Depleted';
+
+                    outputFilename = `${getDungeonByMapId(cm.mapId)} +${cm.level} (${resultText})`;
+                }
+            break;
+        }
+
+        if (!outputFilename) {
+            return;
+        }
+
+        return `{${metadata.category}} ${outputFilename}`;
```

Comment:
Not a big fan of curly braces making it to something user visible. Think it would be better to be:

```
2022-09-22 20-33-47 - 2v2 - Enigma Crucible (Win)
2022-09-22 20-30-39 - Mythic+ - Mechagon Junkyard +18 (+1)
```
Spaces in file names is weird to me as well but I think that's just business as usual on windows so I'll get over it!

Reply:
Spaces in filenames is totally normal on Windows, yeah :D Curly braces look bad, agreed.

Reply:
Spaces in file names are here to stay :p Curly braces looks bad, I agree.